GX-TimeFrequency/src/minimax_omega.F90

```diff
@@ -2199,40 +2198,40 @@ subroutine get_points_weights_omega(grid_size, e_range, ac_we, ierr)
     !> Begin work
     ierr = 0
 
-    if (any(omega_npoints_supported == grid_size)) then
-       ! Find location of grid size
-       kloc = findloc(omega_npoints_supported, grid_size, 1)
-       bup = energy_ranges_grids(kloc)
+    if (.not. any(omega_npoints_supported == grid_size)) then
+       ierr = 1
+       _REGISTER_EXC("The grid size you chose is not available.")
+       return
+    end if
 
-       ! Allocate and set type elements
-       allocate(aw%energy_range(bup))
-       allocate(aw%aw_erange_matrix(2*grid_size, bup+1))
-       call set_aw_array(grid_size, aw)
+    ! Find location of grid size
```

Comment:
From here to the bottom of the routine, this code is identical to what's in minimax_tau.F90's routine. One could reduce this to  


```fortran
subroutine singe get_points_weights_omega(...)

      if (.not. any(omega_npoints_supported == grid_size)) then
        ierr = 1
        _REGISTER_EXC("The grid size you chose is not available.")
        return
     end if

   call get_points_weights(...)
end subroutine
```

and define `get_points_weights` once

Reply:
I will fix this now. Have a look when I push again

Reply:
@AlexBuccheri check it out now. I added a type bound procedure. One could still add allocation and deallocation procedures, the question is how much better readability will be. Still, I agree there is some repetition

Reply:
See my new comments on this change